cmd/pitr/recoverer/recoverer.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return errors.Wrap(err, "get binlog list")
+	}
+	err = r.Recover()
+	if err != nil {
+		return errors.Wrap(err, "recover")
+	}
+	return nil
+}
+
+func (r *Recoverer) Recover() error {
+	host, err := r.getHost()
+	if err != nil {
+		return errors.Wrap(err, "get host")
+	}
+	cfgCmd := exec.Command("sh", "-c", `mc -C `+r.localDest+`mc config host add dest "${ENDPOINT:-https://s3.amazonaws.com}" "$ACCESS_KEY_ID" "$SECRET_ACCESS_KEY"`)
+	err = cfgCmd.Run()
+	if err != nil {
+		return errors.Wrap(err, "update mc config")
+	}
+	flags := ""
+
+	// TODO: add logic for all types
+	switch r.recoverType {
+	case Skip:
+		flags = "--exclude-gtids=" + r.excludingGTIDSet
+	case Transaction:
+	case Date:
+	case Latest:
+	}
+
+	for _, binlog := range r.binlogs {
+		err = r.DownloadBinlog(binlog)
+		if err != nil {
+			return errors.Wrap(err, "download binlog")
+		}
+
+		cmdString := "mc -C " + r.localDest + "mc cat " + r.s3BucketName + "/" + binlog + " | mysqlbinlog " + flags + " - | mysql -h" + host + " -u" + r.pxcUser + " -p$PXC_PASS"
+		cmd := exec.Command("sh", "-c", cmdString)
+		var outb, errb bytes.Buffer
+		cmd.Stdout = &outb
+		cmd.Stderr = &errb
+		err := cmd.Run()
+		if err != nil {
+			return errors.Wrap(err, "run cmd")
+		}
+		if errb.Bytes() != nil {
+			errors.Errorf("cmd error: %s", errb.String())
+		}
+		err = os.Remove(r.localDest + binlog)
+		if err != nil {
+			return errors.Wrap(err, "remove binlog")
+		}
+	}
+
+	return nil
+}
+
+func (r *Recoverer) DownloadBinlog(binlog string) error {
+	content, err := r.storage.GetObject(binlog)
+	if err != nil {
+		return errors.Wrap(err, "get object")
+	}
+	f, err := os.Create(r.localDest + binlog)
+	if err != nil {
+		return errors.Wrap(err, "create file")
+	}
+	defer f.Close()
+	_, err = f.Write(content)
+	if err != nil {
+		return errors.Wrap(err, "write content to file")
+	}
+	return nil
+}
+
+func (r *Recoverer) GetLastBackupGTID() (int64, error) {
+	sep := []byte("GTID of the last")
+
+	o, err := r.storage.GetObject(r.backupNAme + "/xtrabackup_info.lz4.00000000000000000000")
+	if err != nil {
+		return 0, errors.Wrap(err, "get object")
+	}
+
+	startIndex := bytes.Index(o, sep)
+	if startIndex == -1 {
+		return 0, errors.New("no gtid  data in backup")
+	}
+	newOut := o[startIndex+len(sep):]
+	e := bytes.Index(newOut, []byte("'\n"))
+	if e == -1 {
+		return 0, errors.New("cant find gtid  data in backup")
+	}
+
+	set := newOut[:e]
+	setArr := bytes.Split(set, []byte("-"))
+	if len(setArr) < 2 {
+		return 0, errors.New("cant find lastgtid in backup")
+	}
+	idBytes := setArr[len(setArr)-1]
+
+	id, err := strconv.ParseInt(string(idBytes), 10, 64)
+	if err != nil {
+		return 0, errors.New("cant convert last gtid to int64")
+	}
+
+	return id, nil
+}
+
+func (r *Recoverer) SetBinlogs(startID int64) error {
+	saveBinlog := false
+	for _, binlog := range r.storage.ListObjects("") {
+		if strings.Contains(binlog, "binlog.") && !strings.Contains(binlog, "gtid-set") {
```

Comment:
what is the format of gtid-set?
can we make it the part of the binlog name, not to have this extra files?
 Even if we  can't, please document the format